worker/resourceutils.go

```diff
@@ -0,0 +1,84 @@
+package worker
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strconv"
+	"syscall"
+)
+
+const cgroupPath = "/sys/fs/cgroup"
+
+var testmode = false
+
+func addCgroupLimit(jobID string, pid int, maxCPUBandwidth *uint32, maxMemoryUsageBytes *uint64, maxRBPS *uint64, maxWBPS *uint64, maxRIOPS *uint64, maxWIOPS *uint64) error {
+	if testmode {
+		return nil
+	}
+	cgPath := filepath.Join(cgroupPath, jobID)
+	if err := os.MkdirAll(cgPath, 0700); err != nil {
+		return fmt.Errorf("failed to create cgroup: %w", err)
+	}
+
+	if err := ioutil.WriteFile(filepath.Join(cgPath, "cgroup.subtree_control"), []byte("+cpu +memory +io"), syscall.O_WRONLY); err != nil {
```

Comment:
```suggestion
	if err := os.WriteFile(filepath.Join(cgPath, "cgroup.subtree_control"), []byte("+cpu +memory +io"), syscall.O_WRONLY); err != nil {
```

(applies to all the other calls to `ioutil.WriteFile`)